### ISSUE TITLE
Compile errors from OD.h and OD.c due to PDO modification

### ISSUE THREAD

**Issue by woudie** [CANopenNode/CANopenEditor#2]
Hi,

I've been trying to re-map some PDOs using CANopenEdtior, but for some reason, the generated OD.h and OD.c seem to fail when trying to compile with the rest of my source code. After some trial and error, I figured that this occurs only after editing a PDO. The error log is shown in the image. Any advice or assistance on resolving this issue would be greatly appreciated!

![Screenshot 2021-07-12 11:23:22](https://user-images.githubusercontent.com/8084590/125313908-b40ba080-e303-11eb-9b3e-ea5df306a8d9.png)

**Comment by woudie** [comment]
Additionally, it seems that the missing attributes were all SDO permissions

**Comment by woudie** [comment]
Newer update: Most of this problem seems to occur when I try using the TPDO or RPDO tab, but if I use the object dictionary tab to edit the PDOs, it seems mostly fine. I say mostly because the counters of OD objects in the OD.h were still screwed up, but at least I was able to modify the PDOs.

**Comment by nkonidaris** [comment]
I'm facing the same problem, is there any update on a fix for this or have I misconfigured the PDOs ?
![image](https://user-images.githubusercontent.com/63271093/135732818-54bc8dea-97d0-4a7b-b85a-7cf67f1f0485.png)

**Comment by woudie** [comment]
From what I remember, the RPDO and TPDO mapping tabs are completely broken. By that I mean, the generated OD after modifying anything on those tabs will break your OD even if you enter everything correctly. As long as you only use the Object Dictionary tab, your generated OD will be fine. Hopefully somebody comes along and fixes those tabs haha

**Comment by andrew-elder** [comment]
Has this been fixed in mainline code yet?

**Comment by Regelink** [comment]
No, pull request #1 needs to be merged by the maintainer. In the mean time, feel free to use my fork [Regelink/CANopenEditor](https://github.com/Regelink/CANopenEditor)

**Comment by trojanobelix** [comment]
Can someone provide an xdd to reproduce this behavior?

**Comment by CANopenNode** [comment]
I apologize for my inactivity here. I currently can't find my time to be more active. (But at least I receive notifications from now on.)

@trojanobelix is now co-maintainer. Thank you for your recent commits to the repo.

**Comment by trojanobelix** [comment]
TPDO or RPDO tab should be fixed in the bugfix branch. Please test and create a new issue if still something wrong.
